src/entities/DealTokenSwap.ts

```diff
@@ -259,6 +259,48 @@ export class DealTokenSwap implements IDeal {
     return this.dealDocument.votingSummary.primaryDAO.votes.concat(this.dealDocument.votingSummary.partnerDAO?.votes ?? []);
   }
 
+  @computedFrom("dealDocument.votingSummary.primaryDAO.votes")
+  public get primaryDAOVotes(): Array<IVoteInfo> {
+    return this.dealDocument.votingSummary.primaryDAO.votes;
+  }
+
+  @computedFrom("dealDocument.votingSummary.partnerDAO.votes")
+  public get partnerDAOVotes(): Array<IVoteInfo> {
+    return this.dealDocument.votingSummary.partnerDAO?.votes ?? [];
+  }
+
+  @computedFrom("allVotes")
+  public get submittedVotes(): Array<IVoteInfo> {
+    return this.allVotes.filter(voteInfo => voteInfo.vote === true || voteInfo.vote === false);
+  }
+
+  representativeVote(representativeAddress: string): boolean | null {
+    return this.votesArrayToMap(this.allVotes).get(representativeAddress);
+  }
+
+  representativeVoted(representativeAddress: string): boolean {
+    const vote = this.representativeVote(representativeAddress);
+    return vote === true || vote === false;
+  }
+
+  isTheProposalLead(address: string) {
+    return this.registrationData.proposalLead.address === address;
+  }
+
+  representativeDAO(address: string): IDAO | undefined {
```

Comment:
@dkent600 Correct I have implemented this in my branch already, but that branch hasn't been merged to development yet so I'm not sure which one you want to use going foward.

@IonelLupu This is how I implemented it:

![image](https://user-images.githubusercontent.com/2163332/161172724-4bda06ed-e6c0-4ca1-bd58-02bd6d647570.png)

Reply:
@dkent600 I will wait for @vipergtsrz's implementation then.

@dkent600 let's add a lint rule to force us write the `public` method modifier. I usually don't write it because it's `public` by default. A lint rule should help us not forget to write it

Reply:
Can you please copy this code you quoted above from Chad branch into yours?

Reply:
I'll find that lint rule, been wanted to do that for a while.